src/jmeter/java/teammates/lnp/scripts/create/data/CreateStudentProfileTestData.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    @Override
+    protected JSONObject generateAccountsJson() {
+        JSONObject accounts = new JSONObject();
+
+        for (int i = 0; i < NUMBER_OF_USER_ACCOUNTS; i++) {
+            JSONObject studentAccountData = new JSONObject();
+
+            studentAccountData.put("googleId", USER_NAME + i + ".tmms");
+            studentAccountData.put("name", USER_NAME + i);
+            studentAccountData.put("isInstructor", false);
+            studentAccountData.put("email", USER_EMAIL + i + "@gmail.tmt");
+            studentAccountData.put("institute", "TEAMMATES Test Institute 1");
+
+            accounts.put(USER_NAME + i, studentAccountData);
+        }
+
+        JSONObject instructorAccount = new JSONObject();
+        instructorAccount.put("googleId", "TestData.instructor");
+        instructorAccount.put("name", "TEAMMATES Test Instructor");
+        instructorAccount.put("isInstructor", true);
+        instructorAccount.put("email", "tmms.test@gmail.tmt");
+        instructorAccount.put("institute", "TEAMMATES Test Institute 1");
+        accounts.put("instructor", instructorAccount);
+
+        return accounts;
+    }
+
+    @Override
+    protected JSONObject generateCoursesJson() {
+        JSONObject courseData = new JSONObject();
+        courseData.put("id", "TestData.CS101");
+        courseData.put("name", "Intro To Programming");
+        courseData.put("timeZone", "UTC");
+
+        return new JSONObject().put("course", courseData);
+    }
+
+    @Override
+    protected JSONObject generateInstructorsJson() {
+        JSONObject instructors = new JSONObject();
+
+        JSONObject instructorDetails = new JSONObject();
+        instructorDetails.put("googleId", "TestData.instructor");
+        instructorDetails.put("courseId", "TestData.CS101");
+        instructorDetails.put("name", "Teammates Test");
+        instructorDetails.put("email", "tmms.test@gmail.tmt");
+        instructorDetails.put("role", "Co-owner");
+        instructorDetails.put("isDisplayedToStudents", true);
+        instructorDetails.put("displayedName", "Co-owner");
+        instructorDetails.put("sectionLevel", new JSONObject());
+        instructorDetails.put("sessionLevel", new JSONObject());
+
+        JSONObject privileges = new JSONObject();
+        JSONObject courseLevel = new JSONObject();
+        courseLevel.put("canviewstudentinsection", true);
+        courseLevel.put("cansubmitsessioninsection", true);
+        courseLevel.put("canmodifysessioncommentinsection", true);
+        courseLevel.put("canmodifycourse", true);
+        courseLevel.put("canviewsessioninsection", true);
+        courseLevel.put("canmodifysession", true);
+        courseLevel.put("canmodifystudent", true);
+        courseLevel.put("canmodifyinstructor", true);
+        privileges.put("courseLevel", courseLevel);
+
+        instructorDetails.put("privileges", privileges);
+        instructors.put("teammates.test.instructor", instructorDetails);
```

Comment:
Actually why do you need to create these extra info such as instructors, courses, sessions for the student profile test? Is it just to ensure that it works?

Reply:
Yup that's right